orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/ecs/EcsServerGroupCreatorSpec.groovy

```diff
@@ -185,4 +191,235 @@ class EcsServerGroupCreatorSpec extends Specification {
+    mockResolver.getBoundArtifactForStage(stage, testArtifactId, null) >> resolvedArtifact
+    def (testReg,testRepo,testTag) = ["myregistry.io","myrepo","latest"]
+    def testDescription = createTestDescription(testReg, testRepo, testTag)
+    def testMappings = []
+    def map1 = createMap1(testDescription)
+    def map2 = createMap2(testDescription)
+    testMappings.add(map1)
+    testMappings.add(map2)
+
+    // add inputs to stage context
+    stage.execution = new PipelineExecutionImpl(ExecutionType.PIPELINE, 'ecs')
+    stage.context.useTaskDefinitionArtifact = true
+    stage.context.evaluateTaskDefinitionArtifactExpressions = true
+    stage.context.taskDefinitionArtifact = taskDefArtifact
+    stage.context.containerMappings = testMappings
+
+    when:
+    def operations = creator.getOperations(stage)
+
+    then:
+
+    1 * oortService.fetchArtifact(*_) >> new retrofit.client.Response('http://oort.com', 200, 'Okay', [], new TypedString(response))
+    0 * oortService._
+    operations[0].createServerGroup.spelProcessedTaskDefinitionArtifact.toString().equals(expected)
+
+    where:
+    response                                               | expected
+    '{"foo": "bar"}'                                       | "[foo:bar]"
+  }
+
+  def "creates operation when evaluateTaskDefinitionArtifactExpressions is truthy with environment variables"() {
+    given:
+    def testArtifactId = createTaskdefArtifactId()
+    def taskDefArtifact = createTaskDefArtifact(testArtifactId)
+
+    Artifact resolvedArtifact = createResolvedArtifact()
+    mockResolver.getBoundArtifactForStage(stage, testArtifactId, null) >> resolvedArtifact
+    def (testReg,testRepo,testTag) = ["myregistry.io","myrepo","latest"]
+    def testDescription = createTestDescription(testReg, testRepo, testTag)
+    def testMappings = []
+    def map1 = createMap1(testDescription)
+    def map2 = createMap2(testDescription)
+    testMappings.add(map1)
+    testMappings.add(map2)
+
+    // add inputs to stage context
+    stage.execution = new PipelineExecutionImpl(ExecutionType.PIPELINE, 'ecs')
+    stage.context.useTaskDefinitionArtifact = true
+    stage.context.evaluateTaskDefinitionArtifactExpressions = true
+    stage.context.taskDefinitionArtifact = taskDefArtifact
+    stage.context.containerMappings = testMappings
+    stage.execution.trigger.parameters.put("tg", "bar")
+    stage.execution.trigger.parameters.put("ENV1", "bar")
+    stage.context.name = "Deploy"
+
+    when:
+    def operations = creator.getOperations(stage)
+
+    then:
+
+    1 * oortService.fetchArtifact(*_) >> new retrofit.client.Response('http://oort.com', 200, 'Okay', [], new TypedString(response))
+    0 * oortService._
+    operations[0].createServerGroup.spelProcessedTaskDefinitionArtifact.toString().equals(expected)
+
+    where:
+    response                                               | expected
+    '{"foo": "${ENV1}.b.${ENV2}"}'                         | "[foo:\${ENV1}.b.\${ENV2}]"
```

Comment:
I see above you're defining a parameter `ENV1` but then you not referencing it w/ "parameters" syntax here so we're expecting the text as it appears to be resolved instead. Is this understanding of the test correct?

Reply:
So I am basically trying to show here that ENV1 and ENV2 are environment variables which shouldn't be evaluated as we send environment variables and command properties as is to AWS. So instead of having parameter ENV1 available there it is not evaluating that means code knows that ${ENV1} is a env variable and shouldn't be evaluated. Does that make sense?